### ISSUE TITLE
Crash when git is not available

### ISSUE THREAD

**Issue by davidak** [git-learning-game/oh-my-git#16]
I tried to run it on NixOS with `steam-run` to have a common FHS environment for games. It crashes, because there is no `git`.

It starts, i can see the menu, but when i start a level, it crashes.

A solution would be to show a message at start, that git has to be installed on the system. Also, document needed external dependencies. Not sure if it would make sense to include it. When this game is the very first step with git, it might makes sense to not even have to install it. That might be a barrier to some.

```
[davidak@gaming:~/Downloads]$ steam-run ./oh-my-git --verbose
Godot Engine v3.2.3.stable.official - https://godotengine.org
XInput: Refreshing devices.
XInput: No touch devices found.
Custom libGL override detected. Skipping GPU detection
Custom libGL override detected. Skipping GPU detection
Failed loading custom cursor: size_bdiag
Failed loading custom cursor: size_fdiag
Using GLES3 video driver
OpenGL ES 3.0 Renderer: AMD Radeon (TM) RX 480 Graphics (POLARIS10, DRM 3.39.0, 5.9.16, LLVM 9.0.1)
PulseAudio: detected 2 channels
PulseAudio: audio buffer frames: 512 calculated latency: 11ms
 
Loading resource: res://default_bus_layout.tres
CORE API HASH: 0
EDITOR API HASH: 0
Loading resource: res://scenes/helpers.gdc
Loading resource: res://scenes/levels.gdc
Loading resource: res://scenes/chapter.gdc
Loading resource: res://scenes/level.gdc
Loading resource: res://scenes/level_repo.gdc
Loading resource: res://scenes/game.tscn
Loading resource: res://scenes/game.gdc
Loading resource: res://scenes/notification.tscn
Loading resource: res://fonts/default.tres
Loading resource: res://fonts/cabin-regular.ttf
Loading resource: res://styles/theme.tres
Loading resource: res://fonts/monospace.tres
Loading resource: res://fonts/iosevka-regular.ttf
Loading resource: res://scenes/notification.gdc
Loading resource: res://scenes/shell.gdc
Loading resource: res://scenes/shell_command.gdc
Loading resource: res://music/gigantic-greasy-giraffe.ogg
Loading resource: res://scenes/title.tscn
Loading resource: res://images/oh-my-git.png
Loading resource: res://scenes/title.gdc
Loading resource: res://scenes/level_select.tscn
Loading resource: res://scenes/level_select.gdc
Loading resource: res://fonts/big.tres
Loading resource: res://fonts/cabin-bold.ttf
Loading resource: res://scenes/cli_badge.tscn
Loading resource: res://images/cli-badge.svg
Loading resource: res://scenes/cli_badge.gdc
Loading resource: res://styles/alert_button.tres
Loading resource: res://scenes/main.tscn
Loading resource: res://scenes/terminal.tscn
Loading resource: res://sounds/typewriter_ding.wav
Loading resource: res://scenes/terminal.gdc
Loading resource: res://scenes/text_editor.tscn
Loading resource: res://scenes/text_editor.gdc
Loading resource: res://scenes/tcp_server.tscn
Loading resource: res://scenes/tcp_server.gdc
Loading resource: res://sounds/buzzer.wav
Loading resource: res://scenes/main.gdc
Loading resource: res://scenes/repository.tscn
Loading resource: res://scenes/repository.gdc
Loading resource: res://scenes/node.tscn
Loading resource: res://scenes/node.gdc
Loading resource: res://scenes/arrow.tscn
Loading resource: res://scenes/arrow.gdc
Loading resource: res://nodes/blob.svg
Loading resource: res://nodes/tree.svg
Loading resource: res://nodes/commit.svg
Loading resource: res://nodes/ref.svg
Loading resource: res://nodes/head3.svg
Loading resource: res://nodes/pop.wav
Loading resource: res://scenes/drop_area.tscn
Loading resource: res://scenes/drop_area.gdc
Loading resource: res://scenes/cards.tscn
Loading resource: res://scenes/cards.gdc
Loading resource: res://scenes/card.tscn
Loading resource: res://scenes/card.gdc
Loading resource: res://scenes/input_dialog.tscn
Loading resource: res://scripts/input_dialog.gdc
Loading resource: res://scenes/card_particles.tscn
Loading resource: res://sounds/swish.wav
Loading resource: res://sounds/swoosh.wav
Loading resource: res://sounds/poof.wav
Loading resource: res://fonts/small.tres
Loading resource: res://scenes/file_browser.tscn
Loading resource: res://scenes/file_browser.gdc
Loading resource: res://scenes/file_browser_item.tscn
Loading resource: res://scenes/file_browser_item.gdc
Loading resource: res://nodes/document.svg
Loading resource: res://sounds/success.wav
Loading resource: res://scenes/cursor.gdc
ERROR: call: OS.execute failed: bash [-c, "'export HOME='"'"'/home/davidak/.local/share/Oh My Git/tmp/'"'"';export PATH='"'"'/home/davidak/.local/share/Oh My Git/tmp/:/usr/bin:/bin:/mingw64/bin/'"'"';cd '"'"'/home/davidak/.local/share/Oh My Git/tmp/repos/yours/'"'"' || exit 1;git init'"] Output: bash: git: command not found
 
Exit Code 127
   At: modules/gdscript/gdscript_functions.cpp:803.
FATAL ERROR: OS.execute failed: bash [-c, "'export HOME='"'"'/home/davidak/.local/share/Oh My Git/tmp/'"'"';export PATH='"'"'/home/davidak/.local/share/Oh My Git/tmp/:/usr/bin:/bin:/mingw64/bin/'"'"';cd '"'"'/home/davidak/.local/share/Oh My Git/tmp/repos/yours/'"'"' || exit 1;git init'"] Output: bash: git: command not found
 
Exit Code 127
ERROR: call: OS.execute failed: bash [-c, "'export HOME='"'"'/home/davidak/.local/share/Oh My Git/tmp/'"'"';export PATH='"'"'/home/davidak/.local/share/Oh My Git/tmp/:/usr/bin:/bin:/mingw64/bin/'"'"';cd '"'"'/home/davidak/.local/share/Oh My Git/tmp/repos/yours/'"'"' || exit 1;git symbolic-ref HEAD refs/heads/main'"] Output: bash: git: command not found
 
Exit Code 127
   At: modules/gdscript/gdscript_functions.cpp:803.
FATAL ERROR: OS.execute failed: bash [-c, "'export HOME='"'"'/home/davidak/.local/share/Oh My Git/tmp/'"'"';export PATH='"'"'/home/davidak/.local/share/Oh My Git/tmp/:/usr/bin:/bin:/mingw64/bin/'"'"';cd '"'"'/home/davidak/.local/share/Oh My Git/tmp/repos/yours/'"'"' || exit 1;git symbolic-ref HEAD refs/heads/main'"] Output: bash: git: command not found
 
Exit Code 127
WARNING: cleanup: ObjectDB instances leaked at exit (run with --verbose for details).
   At: core/object.cpp:2135.
Leaked instance: Node:3062 - Node name: 
Leaked instance: Node:1453 - Node name: 
Leaked instance: Node:10426 - Node name: 
Leaked instance: Node:5886 - Node name: 
Leaked instance: Node:7139 - Node name: 
Leaked instance: Node:8011 - Node name: 
Leaked instance: Node:4633 - Node name: 
Leaked instance: Node:6202 - Node name: 
Leaked instance: Node:2799 - Node name: 
Leaked instance: Node:3739 - Node name: 
Leaked instance: Node:3853 - Node name: 
Leaked instance: Node:5100 - Node name: 
Leaked instance: Node:8689 - Node name: 
Leaked instance: Node:9425 - Node name: 
Leaked instance: Node:2638 - Node name: 
Leaked instance: Node:4742 - Node name: 
Leaked instance: Node:7019 - Node name: 
Leaked instance: Node:5212 - Node name: 
Leaked instance: Node:7504 - Node name: 
Leaked instance: Node:9374 - Node name: 
Leaked instance: Node:1670 - Node name: 
Leaked instance: Node:8902 - Node name: 
Leaked instance: Node:4956 - Node name: 
Leaked instance: Node:10500 - Node name: 
Leaked instance: Node:4011 - Node name: 
Leaked instance: Node:10445 - Node name: 
Leaked instance: Node:4520 - Node name: 
Leaked instance: Node:6911 - Node name: 
Leaked instance: Node:5942 - Node name: 
Leaked instance: Node:5214 - Node name: 
Leaked instance: GDScript:1153 - Resource path: res://scenes/level.gd
Leaked instance: Node:2015 - Node name: 
Leaked instance: Node:4630 - Node name: 
Leaked instance: Node:8240 - Node name: 
Leaked instance: Node:10446 - Node name: 
Leaked instance: Node:5590 - Node name: 
Leaked instance: Node:5075 - Node name: 
Leaked instance: Node:5220 - Node name: 
Leaked instance: Node:5726 - Node name: 
Leaked instance: Node:1818 - Node name: 
Leaked instance: Node:4852 - Node name: 
Leaked instance: Node:1317 - Node name: 
Leaked instance: Node:6203 - Node name: 
Leaked instance: Node:10498 - Node name: 
Leaked instance: Node:7844 - Node name: 
Leaked instance: GDScript:1154 - Resource path: res://scenes/level_repo.gd
Leaked instance: Node:3852 - Node name: 
Leaked instance: Node:3229 - Node name: 
Leaked instance: Node:6812 - Node name: 
Leaked instance: Node:8381 - Node name: 
Leaked instance: Node:5101 - Node name: 
Leaked instance: Node:2107 - Node name: 
Leaked instance: Node:9353 - Node name: 
Leaked instance: Node:2372 - Node name: 
Leaked instance: Node:8377 - Node name: 
Leaked instance: Node:10407 - Node name: 
Leaked instance: Node:7898 - Node name: 
Leaked instance: Node:3234 - Node name: 
Leaked instance: Node:2206 - Node name: 
Leaked instance: Node:1406 - Node name: 
Leaked instance: Node:2746 - Node name: 
Leaked instance: GDScript:1152 - Resource path: res://scenes/chapter.gd
Leaked instance: Node:6287 - Node name: 
Leaked instance: Node:8688 - Node name: 
Leaked instance: Node:3158 - Node name: 
Leaked instance: Node:6380 - Node name: 
Leaked instance: Node:9015 - Node name: 
Leaked instance: Node:2373 - Node name: 
Leaked instance: Node:4298 - Node name: 
Leaked instance: Node:9373 - Node name: 
Leaked instance: Node:7140 - Node name: 
Leaked instance: Node:4104 - Node name: 
Leaked instance: Node:2641 - Node name: 
Leaked instance: Node:1516 - Node name: 
Leaked instance: Node:3059 - Node name: 
Leaked instance: Node:4108 - Node name: 
Leaked instance: Node:5788 - Node name: 
Leaked instance: Node:10470 - Node name: 
Leaked instance: Node:5219 - Node name: 
Leaked instance: Node:10497 - Node name: 
Leaked instance: Node:7335 - Node name: 
Leaked instance: Node:2014 - Node name: 
Leaked instance: Node:3625 - Node name: 
Leaked instance: Node:8287 - Node name: 
Leaked instance: Node:5885 - Node name: 
Leaked instance: Node:10475 - Node name: 
Leaked instance: Node:3410 - Node name: 
Leaked instance: Node:5210 - Node name: 
Leaked instance: Node:5890 - Node name: 
Leaked instance: Node:2376 - Node name: 
Leaked instance: GDScriptNativeClass:679
Leaked instance: Node:8242 - Node name: 
Leaked instance: Node:1733 - Node name: 
Leaked instance: Node:5216 - Node name: 
Leaked instance: Node:8187 - Node name: 
Leaked instance: Node:5887 - Node name: 
Leaked instance: Node:2205 - Node name: 
Leaked instance: Node:8474 - Node name: 
Leaked instance: Node:3967 - Node name: 
Leaked instance: Node:2745 - Node name: 
Leaked instance: Node:8570 - Node name: 
Leaked instance: Node:3159 - Node name: 
Leaked instance: Node:4743 - Node name: 
Leaked instance: Node:9424 - Node name: 
Leaked instance: Node:9350 - Node name: 
Leaked instance: Node:4415 - Node name: 
Leaked instance: Node:6009 - Node name: 
Leaked instance: Node:2538 - Node name: 
Leaked instance: Node:7507 - Node name: 
Leaked instance: Node:3969 - Node name: 
Leaked instance: Node:3968 - Node name: 
Leaked instance: Node:5725 - Node name: 
Leaked instance: Node:8903 - Node name: 
Leaked instance: Node:2637 - Node name: 
Leaked instance: Node:6914 - Node name: 
Leaked instance: Node:3411 - Node name: 
Leaked instance: Node:6645 - Node name: 
Leaked instance: Node:7502 - Node name: 
Leaked instance: Node:9423 - Node name: 
Leaked instance: Node:5080 - Node name: 
Leaked instance: Node:5151 - Node name: 
Leaked instance: Node:10401 - Node name: 
Leaked instance: Node:10400 - Node name: 
Leaked instance: Node:5077 - Node name: 
Leaked instance: Node:6005 - Node name: 
Leaked instance: Node:7503 - Node name: 
Leaked instance: Node:5076 - Node name: 
Leaked instance: Node:10495 - Node name: 
Leaked instance: Node:7946 - Node name: 
Leaked instance: Node:3672 - Node name: 
Leaked instance: Node:1612 - Node name: 
Leaked instance: Node:4201 - Node name: 
Leaked instance: Node:1613 - Node name: 
Leaked instance: Node:2947 - Node name: 
Leaked instance: Node:8124 - Node name: 
Leaked instance: Node:10406 - Node name: 
Leaked instance: Node:9125 - Node name: 
Leaked instance: Node:2866 - Node name: 
Leaked instance: Node:2800 - Node name: 
Leaked instance: Node:1732 - Node name: 
Leaked instance: Node:6288 - Node name: 
Leaked instance: Node:3571 - Node name: 
Leaked instance: Node:1736 - Node name: 
Leaked instance: Node:3572 - Node name: 
Leaked instance: Node:8183 - Node name: 
Leaked instance: Node:6811 - Node name: 
Leaked instance: Node:7945 - Node name: 
Leaked instance: Node:9483 - Node name: 
Leaked instance: Node:7845 - Node name: 
Leaked instance: Node:5789 - Node name: 
Leaked instance: Node:7220 - Node name: 
Leaked instance: Node:9016 - Node name: 
Leaked instance: Node:3230 - Node name: 
Leaked instance: Node:4519 - Node name: 
Leaked instance: Node:1405 - Node name: 
Leaked instance: Node:7683 - Node name: 
Leaked instance: Node:10448 - Node name: 
Leaked instance: Node:4853 - Node name: 
Leaked instance: Node:5679 - Node name: 
Leaked instance: Node:8906 - Node name: 
Leaked instance: Node:3911 - Node name: 
Leaked instance: Node:1617 - Node name: 
Leaked instance: Node:4957 - Node name: 
Leaked instance: Node:1226 - Node name: 
Leaked instance: Node:9126 - Node name: 
Leaked instance: Node:10450 - Node name: 
Leaked instance: Node:4416 - Node name: 
Leaked instance: Node:6006 - Node name: 
Leaked instance: Node:6649 - Node name: 
Leaked instance: Node:1614 - Node name: 
Leaked instance: Node:6478 - Node name: 
Leaked instance: Node:1229 - Node name: 
Leaked instance: Node:9230 - Node name: 
Leaked instance: Node:6379 - Node name: 
Leaked instance: Node:10471 - Node name: 
Leaked instance: Node:8792 - Node name: 
Leaked instance: Node:3058 - Node name: 
Leaked instance: Node:7332 - Node name: 
Leaked instance: Node:3914 - Node name: 
Leaked instance: Node:5152 - Node name: 
Leaked instance: Node:5150 - Node name: 
Leaked instance: Node:10399 - Node name: 
Leaked instance: Node:7018 - Node name: 
Leaked instance: Node:8475 - Node name: 
Leaked instance: Node:3738 - Node name: 
Leaked instance: Node:6091 - Node name: 
Leaked instance: Node:8241 - Node name: 
Leaked instance: Node:10402 - Node name: 
Leaked instance: Node:9348 - Node name: 
Leaked instance: Node:2539 - Node name: 
Leaked instance: Node:3624 - Node name: 
Leaked instance: Node:4202 - Node name: 
Leaked instance: Node:7072 - Node name: 
Leaked instance: Node:1930 - Node name: 
Leaked instance: Node:4297 - Node name: 
Leaked instance: Node:4629 - Node name: 
Leaked instance: Node:7073 - Node name: 
Leaked instance: Node:8378 - Node name: 
Leaked instance: Node:5211 - Node name: 
Leaked instance: Node:3910 - Node name: 
Leaked instance: Node:1669 - Node name: 
Leaked instance: Node:7848 - Node name: 
Leaked instance: Node:10476 - Node name: 
Leaked instance: Node:8571 - Node name: 
Leaked instance: Node:4014 - Node name: 
Leaked instance: Node:1452 - Node name: 
Leaked instance: Node:6910 - Node name: 
Leaked instance: Node:2946 - Node name: 
Leaked instance: GDScript:1196 - Resource path: res://scenes/shell_command.gd
Leaked instance: Node:7431 - Node name: 
Leaked instance: Node:5502 - Node name: 
Leaked instance: Node:7432 - Node name: 
Leaked instance: Node:3160 - Node name: 
Leaked instance: Node:8184 - Node name: 
Leaked instance: Node:8793 - Node name: 
Leaked instance: Node:10447 - Node name: 
Leaked instance: GDScript:1195 - Resource path: res://scenes/shell.gd
Leaked instance: Node:5499 - Node name: 
Leaked instance: Node:7433 - Node name: 
Leaked instance: Node:6646 - Node name: 
Leaked instance: Node:1819 - Node name: 
Leaked instance: Node:8126 - Node name: 
Leaked instance: Node:10398 - Node name: 
Leaked instance: Node:2106 - Node name: 
Leaked instance: Node:7897 - Node name: 
Leaked instance: Node:1929 - Node name: 
Leaked instance: Node:8284 - Node name: 
Leaked instance: Node:3575 - Node name: 
Leaked instance: Node:1515 - Node name: 
Leaked instance: Node:6092 - Node name: 
Leaked instance: Node:8012 - Node name: 
Leaked instance: Node:9349 - Node name: 
Leaked instance: Node:10499 - Node name: 
Leaked instance: Node:5678 - Node name: 
Leaked instance: Node:7684 - Node name: 
Leaked instance: Node:5943 - Node name: 
Leaked instance: Node:6479 - Node name: 
Leaked instance: Node:9229 - Node name: 
Leaked instance: Node:10496 - Node name: 
Leaked instance: Node:7219 - Node name: 
Leaked instance: Node:7331 - Node name: 
Leaked instance: Node:10449 - Node name: 
Leaked instance: Node:3673 - Node name: 
Leaked instance: Node:8125 - Node name: 
Leaked instance: Node:1316 - Node name: 
Leaked instance: Node:3231 - Node name: 
Leaked instance: Node:5217 - Node name: 
Leaked instance: Node:3851 - Node name: 
Leaked instance: Node:5589 - Node name: 
Leaked instance: Node:4010 - Node name: 
Leaked instance: Node:2867 - Node name: 
Leaked instance: Node:8283 - Node name: 
Leaked instance: Node:4105 - Node name: 
Hint: Leaked instances typically happen when nodes are removed from the scene tree (with `remove_child()`) but not freed (with `free()` or `queue_free()`).
ERROR: clear: Resources still in use at exit (run with --verbose for details).
   At: core/resource.cpp:477.
Resource still in use: res://scenes/shell.gd (GDScript)
Resource still in use: res://scenes/level.gd (GDScript)
Resource still in use: res://scenes/level_repo.gd (GDScript)
Resource still in use: res://scenes/shell_command.gd (GDScript)
Resource still in use: res://scenes/chapter.gd (GDScript)
Orphan StringName: compile
Orphan StringName: run_async
Orphan StringName: push_back
Orphan StringName: _unused
Orphan StringName: read_file
Orphan StringName: global_shell
Orphan StringName: res://scenes/level.gd
Orphan StringName: _shell_binary
Orphan StringName: tipp_level
Orphan StringName: res://scenes/shell.gd
Orphan StringName: res://scenes/chapter.gd
Orphan StringName: _to_string
Orphan StringName: construct
Orphan StringName: get_available_bytes
Orphan StringName: new
Orphan StringName: levels
Orphan StringName: exit_code
Orphan StringName: GDScriptNativeClass
Orphan StringName: congrats
Orphan StringName: HOME
Orphan StringName: get_utf8_string
Orphan StringName: _os
Orphan StringName: check_win
Orphan StringName: split
Orphan StringName: path
Orphan StringName: get_name
Orphan StringName: strip_edges
Orphan StringName: sort
Orphan StringName: res://scenes/level_repo.gd
Orphan StringName: get
Orphan StringName: begins_with
Orphan StringName: repos
Orphan StringName: run
Orphan StringName: cd
Orphan StringName: open
Orphan StringName: description
Orphan StringName: careful_delete
Orphan StringName: action_commands
Orphan StringName: read_from
Orphan StringName: res://scenes/shell_command.gd
Orphan StringName: get_next
Orphan StringName: append
Orphan StringName: crash
Orphan StringName: command
Orphan StringName: setup_commands
Orphan StringName: emit_signal
Orphan StringName: keys
Orphan StringName: start
Orphan StringName: list_dir_end
Orphan StringName: slug
Orphan StringName: erase
Orphan StringName: title
Orphan StringName: has
Orphan StringName: Node
Orphan StringName: length
Orphan StringName: write_file
Orphan StringName: file_exists
Orphan StringName: sub
Orphan StringName: done
Orphan StringName: _cwd
Orphan StringName: crash_on_fail
Orphan StringName: list_dir_begin
Orphan StringName: cards
Orphan StringName: PATH
Orphan StringName: output
Orphan StringName: parse
Orphan StringName: load
Orphan StringName: exec
Orphan StringName: _init
Orphan StringName: substr
Orphan StringName: size
Orphan StringName: replace
Orphan StringName: GDScript
Orphan StringName: win_conditions
Orphan StringName: run_async_thread
Orphan StringName: tmp_prefix
Orphan StringName: thread
StringName: 77 unclaimed string names at exit.
ERROR: cleanup: Condition "allocs_used > 0" is true.
   At: core/pool_vector.cpp:69.
```

**Comment by blinry** [comment]
Good point, we should definitely catch this, and give a nice error message! :) Thanks!

**Comment by blinry** [comment]
Should be fixed with 2fc78f3 and 9d8c8e2! :)
